docs/workflow/building/coreclr/README.md

```diff
@@ -1,56 +1,100 @@
+Specifying `-subset` explicitly is not necessary if it is the first argument: `./build.sh --subset clr` and `./build.sh clr` are equivalent. However, if you specify any other argument beforehand, then you must specify the `-subset` flag.
+
+## Common Building Options
+
+By default, the script generates a _Debug_ build type, which is not optimized code and includes asserts. As its name suggests, this makes it easier and friendlier to debug the code. If you want to make performance measurements, you ought to build the _Release_ version instead, which doesn't have any asserts and has all code optimizations enabled. Likewise, if you plan on running tests, the _Release_ configuration is more suitable since it's considerably faster than the _Debug_ one. For this, you add the flag `-configuration release` (or `-c release`). For example:
+
+```bash
+./build.sh --subset clr --configuration release
 ```
-./build.sh -subset clr -configuration checked
+
+As mentioned before in the [general building document](/docs/workflow/README.md#configurations-and-subsets), CoreCLR also supports a _Checked_ build type which has asserts enabled like _Debug_, but is built with the native compiler optimizer enabled, so it runs much faster. This is the usual mode used for running tests in the CI system.
+
+Now, it is also possible to select a different configuration for each subset when building them together. The `--configuration` flag applies universally to all subsets, but it can be overridden with any one or more of the following ones:
+
+* `--runtimeConfiguration (-rc)`: Flag for the CLR build configuration.
+* `--librariesConfiguration (-lc)`: Flag for the libraries build configuration.
+* `--hostConfiguration (-hc)`: Flag for the host build configuration.
+
+For example, a very common scenario used by developers is to build the _clr_ in _Debug_ mode, and the _libraries_ in _Release_ mode. To achieve this, the command-line would look like the following:
+
+```bash
+./build.sh --subset clr+libs --configuration Release --runtimeConfiguration Debug
 ```
 
-If you want to use Ninja to drive the native build instead of Make on non-Windows platforms, you can pass the `-ninja` flag to the build script as follows:
+Or alternatively:
+
+```bash
+./build.sh --subset clr+libs --librariesConfiguration Release --runtimeConfiguration Debug
 ```
-./build.cmd -subset clr -ninja
+
+For more information about all the different options available, supply the argument `-help|-h` when invoking the build script. On Unix-like systems, non-abbreviated arguments can be passed in with a single `-` or double hyphen `--`.
+
+### Build Drivers
+
+If you want to use _Ninja_ to drive the native build instead of _Make_ on non-Windows platforms, you can pass the `-ninja` flag to the build script as follows:
```

Comment:
If you want to be exhaustive in your update, please sync up with @jkoritzinsky as I believe Ninja is now the default and you have to explicitly suppress it if you're not impressed, it's just pre-existing debt in the docs, it wasn't updated after the switchover.

Reply:
We never moved to Ninja as the default for non-Windows. We've discussed it (as well as possibly doing what Mono does and defaulting to Ninja with a silent fallback to Make if Ninja isn't supported on non-windows), but we haven't actually done that.